### PR TITLE
boards: nrf52840_mdk: Added openocd runner support

### PR DESCRIPTION
**Pull request by sdalu**
Added openocd runner with cmsis-dap interface.

Signed-off-by: Stephane D'Alu <sdalu@sdalu.com>

### CODE REVIEW

File: boards/arm/nrf52840_mdk/board.cmake [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 board_runner_args(pyocd "--target=nrf52840")
 include(${ZEPHYR_BASE}/boards/common/pyocd.board.cmake)
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
```
Comment (mbolivar-nordic) [inline boards/arm/nrf52840_mdk/board.cmake:5]: Are you sure about this? Why not use openocd-nrf5.board.cmake instead? That would get you thread-aware debugging for free.

Comment (mbolivar-nordic) [reply boards/arm/nrf52840_mdk/board.cmake:5]: Ah, I see, it's a different interface. Never mind. No objections to this patch, although you could probably adapt openocd-nrf5.board.cmake to get thread aware debugging with your interface as well. see the `pre_init_cmds` variable set in that file for details.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by sdalu** [comment]
See #29286 instead

**Comment by mbolivar-nordic** [comment]
> See #29286 instead

Thanks for this nice improvement.

For future reference, this project prefers force-pushing into the same PR over closing PRs and reopening them when responding to review comments. This preserves history. See the contribution guide for details.

**Pull request closed without merging**